packages/frontend/src/components/common/Filters/renderFilterList.tsx

```diff
@@ -61,40 +68,57 @@ const StickyMenuDivider = forwardRef<HTMLLIElement, StickyMenuDividerProps>(
 const getItemGroupLabel = (item: Field | TableCalculation) =>
     isField(item) ? item.tableLabel : 'Table Calculations';
 
-const renderFilterList = <T extends Field | TableCalculation>(
-    itemListRendererProps: ItemListRendererProps<T>,
-    tables: Pick<SummaryExplore, 'name' | 'description'>[],
-) => {
-    const { items, itemsParentRef, renderItem } = itemListRendererProps;
-    const getGroupedItems = (filteredItems: typeof items) => {
-        return filteredItems.reduce<
-            {
-                group: typeof tables[0];
-                items: typeof items;
-            }[]
-        >((acc, item) => {
-            const table = isField(item)
-                ? tables.find((t) => t.name === item.table)
-                : undefined;
-
-            const group = {
-                name: getItemGroupLabel(item),
-                description: table?.description,
-            };
-
-            const lastGroup = acc[acc.length - 1];
-            if (lastGroup && lastGroup.group.name === group.name) {
-                lastGroup.items.push(item);
-            } else {
-                acc.push({ group, items: [item] });
-            }
-            return acc;
-        }, []);
-    };
+const useGroupedItems = <T extends Field | TableCalculation>(
+    items: T[],
+    tables: GroupTable[],
+): GroupedItems<T>[] => {
+    const tableMap = useMemo(
+        () => new Map(tables.map((table) => [table.name, table])),
+        [tables],
+    );
+
+    const getGroupedItems = useCallback(
+        (filteredItems: typeof items): GroupedItems<T>[] => {
+            return filteredItems.reduce<GroupedItems<T>[]>((acc, item) => {
+                const table = isField(item)
+                    ? tableMap.get(item.table)
+                    : undefined;
+
+                const group = {
+                    name: getItemGroupLabel(item),
+                    description: table?.description,
+                };
+
+                const lastGroup = acc[acc.length - 1];
+                if (lastGroup && lastGroup.group.name === group.name) {
+                    lastGroup.items.push(item);
+                } else {
+                    acc.push({ group, items: [item] });
+                }
+                return acc;
+            }, []);
+        },
+        [tableMap],
+    );
+
+    return useMemo(() => getGroupedItems(items), [getGroupedItems, items]);
+};
+
+type RenderFilterListProps<T> = ItemListRendererProps<T> & {
+    tables: Pick<SummaryExplore, 'name' | 'description'>[];
+};
+
+const RenderFilterList = <T extends Field | TableCalculation>({
+    items,
+    itemsParentRef,
+    renderItem,
+    tables,
+}: RenderFilterListProps<T>) => {
+    const groupedItems = useGroupedItems<T>(items, tables);
```

Comment:
What does making it a hook get you? And easy place to put useMemo?

Reply:
Great Q, I extracted it into its own hook just to separate the rendering logic from the formatting logic. 
happy to revert and keep it all inside the component 😄

Reply:
No, don't need a change, just wondering. I don't think theres a relevant perf impact.